p2p/net/conn/transport.go

```diff
@@ -0,0 +1,191 @@
+package conn
+
+import (
+	"net"
+
+	ma "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr"
+	manet "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net"
+	mautp "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net/utp"
+	reuseport "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-reuseport"
+)
+
+type Transport interface {
+	manet.Listener
```

Comment:
We might want to be able to have a transport option to dial to a given peer, but not be available to listen on that transport, it doesn't have to be a full buy in every time